medusa/indexers/tvmaze/tvmaze_api.py

```diff
@@ -465,3 +465,30 @@ def get_last_updated_series(self, from_time, weeks=1, filter_show_list=None):
             updates = new_list
 
         return updates
+
+    def get_id_by_external(self, **kwargs):
+        """Search tvmaze for a show, using an external id.
+
+        Accepts as kwargs, so you'l need to add the externals as key/values.
+        :param tvrage: The tvrage id.
+        :param thetvdb: The tvdb id.
+        :param imdb: An imdb id (inc. tt).
+        :returns: A dict with externals, including the tvmaze id.
+        """
+        mapping = {'thetvdb': 'tvdb_id', 'tvrage': 'tvrage_id', 'imdb': 'imdb_id'}
+        externals = {}
+        for external_id in ['tvdb_id', 'imdb_id', 'tvrage_id']:
+            if kwargs.get(external_id):
+                try:
+                    result = self.tvmaze_api.get_show(**{external_id: kwargs.get(external_id)})
+                    if result:
+                        externals = result.externals
+                        externals[external_id] = result.id
+                        return {mapping[external_id]: external_value
+                                for external_id, external_value
+                                in externals.items()
+                                if external_value and mapping.get(external_id)}
+                except ShowNotFound:
```

Comment:
fya, I made a small change here. Of for one key are reverse external can't be found, it will try the next one now. Before that I exited. But a show can have for ex a imdb mapped but not a tvdb.